server/server.go

```
@@ -0,0 +1,144 @@
+package server
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"net"
+	"strings"
+	"sync"
+)
+
+var (
+	message        string
+	lock           sync.Mutex
+	maxMessageSize int = 10 * (1 << 10)
+)
+
+type Server struct {
+	host string
+	port string
+}
+
+type Client struct {
+	conn net.Conn
+}
+
+type Config struct {
+	Host string
+	Port string
+}
+
+// new creates a server.
+func NewServer(config *Config) *Server {
+	return &Server{
+		host: config.Host,
+		port: config.Port,
+	}
+}
+
+// run starts the server and start listens for connections.
+func (server *Server) Run() {
+	listener, err := net.Listen("tcp", fmt.Sprintf("%s:%s", server.host, server.port))
+	if err != nil {
+		log.Printf("Error accepting connection: %v", err)
```

Maybe use log.fatal?
Just returning and exiting will return an OK status code to the operating system

---

You can also return an error instead